internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -131,6 +139,46 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 	return ctrl.Result{}, nil
 }
 
+func reconcileManagedLabels(machine *clusterv1.Machine, node *corev1.Node) bool {
+	// Reconcile managed Labels.
+	var labelsHaveChanged bool
```

Comment:
I think with SSA this should be simpler, essentially we should only have to set our desiredLabels in the node object and that's it

Unfortunately this will make it hard to optimize that the SSA is not run on every reconcile. But the alternative is re-implementing the SSA logic on client-side (including that we have to consider ownership of labels)

Reply:
I'll re-write based on the current state